_posts/2025-04-28-running-at-scale.adoc

```diff
@@ -0,0 +1,83 @@
+
+To optimize etcd performance at scale, several best practices should be followed. Using low-latency storage, such as SSDs or NVMe drives on dedicated disks, is crucial for fast read and write operations. Proper etcd cluster sizing, typically with an odd number of members like 3 or 5, helps maintain quorum and fault tolerance without excessive replication overhead. Ensuring that etcd nodes have sufficient CPU and memory resources is also essential for stable operation. Continuously monitoring etcd's health and performance by tracking key metrics such as latency, throughput, and database size allows for early detection of potential issues. Regular maintenance, including compaction and defragmentation of the etcd database, helps reclaim storage space and improve performance. In very large clusters with high pod churn, considering a separate dedicated etcd instance specifically for Kubernetes Event objects can help alleviate the load on the main etcd cluster. Optimizing the way etcd clients, such as the API server, use etcd's API by avoiding large values in Put calls and minimizing frequent updates can also contribute to better performance. Finally, tuning etcd's configuration parameters, such as the database quota (--quota-backend-bytes), heartbeat interval (--heartbeat-interval), and leader election timeout (--election-timeout), might be necessary to tailor its behavior to the specific workload. Given the anticipated high volume of custom resources for CDE workspaces, a proactive approach to managing etcd's data footprint is essential, including regular monitoring of etcd object sizes per type to prevent control plane instability.
```

Comment:
> To optimize etcd performance at scale, several best practices should be followed.

There are a lot of different best practices mentioned in this paragraph, WDYT about formatting them in a bullet point list?

ex:
* Using low-latency storage, such as SSDs or NVMe drives on dedicated disks, is crucial for fast read and write operations.
* Proper etcd cluster sizing, typically with an odd number of members like 3 or 5, helps maintain quorum and fault tolerance without excessive replication overhead.
* Ensuring that etcd nodes have sufficient CPU and memory resources is also essential for stable operation.
* Continuously monitoring etcd’s health and performance by tracking key metrics such as latency, throughput, and database size allows for early detection of potential issues.